src/containers/pages/FocusInvestigation/map/active/index.tsx

```diff
@@ -126,21 +142,62 @@ class SingleActiveFIMap extends React.Component<
       fetchGoalsActionCreator,
       fetchJurisdictionsActionCreator,
       fetchPlansActionCreator,
+      fetchStructuresActionCreator,
       fetchTasksActionCreator,
+      plan,
     } = this.props;
-
-    await supersetFetch(SUPERSET_JURISDICTIONS_SLICE).then((result: Jurisdiction[]) =>
-      fetchJurisdictionsActionCreator(result)
-    );
-    await supersetFetch(SUPERSET_PLANS_SLICE).then((result2: Plan[]) =>
-      fetchPlansActionCreator(result2)
-    );
-    await supersetFetch(SUPERSET_GOALS_SLICE).then((result3: Goal[]) =>
-      fetchGoalsActionCreator(result3)
-    );
-    await supersetFetch(SUPERSET_STRUCTURES_SLICE, { row_limit: 3000 }).then((result4: Task[]) =>
-      fetchTasksActionCreator(result4)
-    );
+    const planId = plan && plan.plan_id;
+    if (planId) {
+      const pivotParams = superset.getFormData(3000, [
+        { comparator: planId, operator: '==', subject: 'plan_id' },
+      ]);
+      let sqlFilterExpression = '';
+      await supersetFetch(SUPERSET_PLAN_STRUCTURE_PIVOT_SLICE, pivotParams).then(
+        async relevantJurisdictions => {
+          if (!relevantJurisdictions) {
+            return new Promise(reject => {
+              reject();
+            });
+          }
+          for (let i = 0; i < relevantJurisdictions.length; i += 1) {
+            const jurId = relevantJurisdictions[i].jurisdiction_id;
+            if (i) {
+              sqlFilterExpression += ' OR ';
+            }
+            sqlFilterExpression += `${JURISDICTION_ID} = '${jurId}'`;
+          }
+          const planJurisdictionSupersetParams = superset.getFormData(1000, [
+            { sqlExpression: sqlFilterExpression },
+          ]);
+          const jurisdictionResults = await supersetFetch(
+            SUPERSET_JURISDICTIONS_SLICE,
+            planJurisdictionSupersetParams
+          );
+          fetchJurisdictionsActionCreator(jurisdictionResults);
+        }
+      );
+      /** define superset params for structures */
+      const structuresparams = superset.getFormData(3000, [{ sqlExpression: sqlFilterExpression }]);
```

Comment:
Why not use an `ad hoc filter`?  I think we need to keep some kind of consistency in how we do the filtering - I had imagined this would be that we only use sql expressions when that is necessary, otherwise we always use the `ad hoc filters`.  But I dont mind what we specifically do, as long as we do it consistently the same way in our codebase.